### PR TITLE
feat(watermark): support for setting text alignment direction through textAlign

### PR DESCRIPTION
**Pull request by Yuiai01**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/44885
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | `watermark` supports setting text alignment direction through `textAlign`          |
| 🇨🇳 Chinese | 水印支持通过 `textAlign` 设置文本对齐方向          |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 5681b65</samp>

Added a `textAlign` prop to the `Watermark` component to enable custom text alignment for the watermark. Updated the demo, the documentation, and the `useClips` hook to support the new prop. Modified the files `components/watermark/demo/custom.tsx`, `components/watermark/index.tsx`, `components/watermark/index.en-US.md`, `components/watermark/index.zh-CN.md`, and `components/watermark/useClips.ts`.

### <samp>🤖 Generated by Copilot at 5681b65</samp>

*  Add `textAlign` prop to `Watermark` component and demo to allow users to customize the horizontal alignment of the watermark text ([link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7L2-R12), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7R21), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7R28-R33), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7R40), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7L28-R46), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7R54), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7R124-R126), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-f338d58197c4270e4dded5a90516016afd08271c7cd24a3c57cd4d5e6613f317R55), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-add08763825dd36ecf1bf3482c1f2e331f8b6c9d83eff23a5fd55a2a25fd2e19R26), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-add08763825dd36ecf1bf3482c1f2e331f8b6c9d83eff23a5fd55a2a25fd2e19R71), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-add08763825dd36ecf1bf3482c1f2e331f8b6c9d83eff23a5fd55a2a25fd2e19R175), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-add08763825dd36ecf1bf3482c1f2e331f8b6c9d83eff23a5fd55a2a25fd2e19R238), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-877ffb6be3567430f8b12395585cc902eca9d2b7a0cd56e032a10bb28e105ad7R57), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-e74de8cfb7d6f9243c101cce673937f42777adc732547f8ef936d6ae1a92e001L51-R56))
*  Set the default value of `textAlign` to `center` in `Watermark` component and demo ([link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7R40), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-add08763825dd36ecf1bf3482c1f2e331f8b6c9d83eff23a5fd55a2a25fd2e19R71))
*  Pass `textAlign` prop to `getClips` function and `useClips` hook in `Watermark` component to generate and render the canvas elements for the watermark ([link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-add08763825dd36ecf1bf3482c1f2e331f8b6c9d83eff23a5fd55a2a25fd2e19R175), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-add08763825dd36ecf1bf3482c1f2e331f8b6c9d83eff23a5fd55a2a25fd2e19R238), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-e74de8cfb7d6f9243c101cce673937f42777adc732547f8ef936d6ae1a92e001L51-R56))
*  Update the documentation for `Watermark` component in English and Chinese versions to include the `textAlign` prop in the `Font` section ([link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-f338d58197c4270e4dded5a90516016afd08271c7cd24a3c57cd4d5e6613f317R55), [link](https://github.com/ant-design/ant-design/pull/44888/files?diff=unified&w=0#diff-877ffb6be3567430f8b12395585cc902eca9d2b7a0cd56e032a10bb28e105ad7R57))

### CODE REVIEW

File: components/watermark/index.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ export interface WatermarkProps {
     fontWeight?: 'normal' | 'light' | 'weight' | number;
     fontStyle?: 'none' | 'normal' | 'italic' | 'oblique';
     fontFamily?: string;
+    textAlign?: 'left' | 'center' | 'right';
```
Comment (li-jia-nan) [inline components/watermark/index.tsx:26]: 是不是可以再加两个CSS逻辑属性 `start` 和 `end` ？

Comment (li-jia-nan) [reply components/watermark/index.tsx:26]: 还有 `justify` 两端对齐

Comment (Yuiai01) [reply components/watermark/index.tsx:26]: 确实可以，不过这样一来就有点太多了 emmmm

Comment (li-jia-nan) [reply components/watermark/index.tsx:26]: @MadCcc 问问大佬意见

Comment (MadCcc) [reply components/watermark/index.tsx:26]: 这个没有现成的类型么

Comment (li-jia-nan) [reply components/watermark/index.tsx:26]: 可以的 `React.CSSProperties['textAlign']`

Comment (li-jia-nan) [reply components/watermark/index.tsx:26]: image

Comment (Yuiai01) [reply components/watermark/index.tsx:26]: 好，那我这边改一下，那这个 demo 就删除了。

Comment (li-jia-nan) [reply components/watermark/index.tsx:26]: 那就用这个类型好了

Comment (Yuiai01) [reply components/watermark/index.tsx]: 好，改好了～

**Review by li-jia-nan — APPROVED** [review]

File: components/watermark/index.en-US.md [hunk 2]
```diff
@@ -52,6 +52,7 @@ Common props ref：[Common props](/docs/react/common-props)
 | fontWeight | font weight | `normal` \| `light` \| `weight` \| number | normal |  |
 | fontFamily | font family | string | sans-serif |  |
 | fontStyle | font style  | `none` \| `normal` \| `italic` \| `oblique` | normal |  |
+| textAlign | specify the text alignment direction  | `left` \| `start` \| `center` \| `right` \| `end` | `center` | 5.10.0 |
```
Comment (MadCcc) [inline components/watermark/index.en-US.md:55]: 这里类型也改一下吧

File: components/watermark/index.tsx [hunk 3]
```diff
@@ -23,6 +23,7 @@ export interface WatermarkProps {
     fontWeight?: 'normal' | 'light' | 'weight' | number;
     fontStyle?: 'none' | 'normal' | 'italic' | 'oblique';
     fontFamily?: string;
+    textAlign?: CanvasTextAlign;
```
[the remark this answers is not in the document]
Comment (Yuiai01) [reply components/watermark/index.tsx:26]: 确实，我来改一下。

File: components/watermark/index.en-US.md (hunk 2 above)
Comment (Yuiai01) [reply components/watermark/index.en-US.md:55]: 就写 CanvasTextAlign 吗

File: components/watermark/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Yuiai01) [reply components/watermark/index.tsx:26]: @MadCcc 闲夕大佬这个属性要不要也给外部提供一下？
image

[the remark this answers is not in the document]
Comment (MadCcc) [reply components/watermark/index.tsx:26]: 有需要再加吧

[the remark this answers is not in the document]
Comment (Yuiai01) [reply components/watermark/index.tsx:26]: 上面的类型只有 color 可以用 CanvasFillStrokeStyles['fillStyle'] 替代，其他的都不行。

**Review by MadCcc — APPROVED** [review]

**Pull request merged**